pyprocessing/math.py

```diff
@@ -162,191 +176,199 @@ def from_angle(angle, target=None):
     def __sub__(self, other):
-        if isinstance(other, (list, tuple)):
-            return self.sub(*other)
-        elif isinstance(other, (PVector, complex)):
-            return self.sub(other)
+        if isinstance(other, complex):
+            other = (other.real, other.imag)
+
+        if hasattr(other, '__iter__'):
+            return PVector(*(a - b for a, b in zip(self, other)))
+
         raise TypeError(
             f"Invalid operand '-' for type PVector and {type(other)}"
         )
 
     def __mul__(self, other):
-        if isinstance(other, PVector):
+        if isinstance(other, (float, int)):
+            return PVector(*(a * other for a in self))
+
+        elif hasattr(other, '__iter__'):
             return self.dot(other)
-        elif isinstance(other, (list, tuple)):
-            return self.dot(PVector(*other))
-        elif isinstance(other, (float, int)):
-            return self.mult(other)
+
         raise TypeError(
             f"Invalid operand '*' for type PVector and {type(other)}"
         )
 
     def __truediv__(self, other):
         if isinstance(other, (float, int)):
-            return self.div(other)
-        raise TypeError(
-            f"Invalid operand '/' for type PVector and {type(other)}"
-        )
+            return PVector(*(a / other for a in self))
 
-    def __div__(self, other):
-        if isinstance(other, (float, int)):
-            return self.div(other)
         raise TypeError(
             f"Invalid operand '/' for type PVector and {type(other)}"
         )
 
     def __floordiv__(self, other):
         if isinstance(other, (float, int)):
-            return self.div(other)
+            return PVector(*(a // other for a in self))
+
         raise TypeError(
             f"Invalid operand '//' for type PVector and {type(other)}"
         )
 
     def __matmul__(self, other):
-        if isinstance(other, PVector):
+        if hasattr(other, '__iter__'):
             return self.cross(other)
-        raise TypeError(
-            f"Invalid operand '@' for type PVector and {type(other)}"
-        )
 
     def __str__(self):
-        if self.is_2d:
-            return f'{self.x} {self.y}'
-        return f'{self.x} {self.y} {self.z}'
+        return str(tuple(self.seq))
 
     def __repr__(self):
-        return f'PVector<{", ".join(str(self).split())}> at {id(self)}'
+        return f"PVector{tuple(self.seq)}"
+
+    def __len__(self):
+        return len(self.seq)
+
+    def __iter__(self):
+        return self.seq.__iter__()
 
     def __eq__(self, other):
-        if isinstance(other, PVector):
-            return all(
-                m1 == m2 for m1, m2 in zip(self.position, other.position)
-            )
-        if isinstance(other, (list, tuple)):
-            return all(m1 == m2 for m1, m2 in zip(self.position, other))
-        raise ValueError(
-            f"Invalid operand '==' for type Pvector and {type(other)}"
-        )
+        if not isinstance(other, PVector):
```

Comment:
PVector(0, 0, 0) != (0, 0, 0) with this implementation, which is a handy shorthand.